e2e/common_test.go

```diff
@@ -0,0 +1,136 @@
+package e2e_test
+
+import (
+	"github.com/open-feature/go-sdk/pkg/openfeature"
+	mp "github.com/open-feature/go-sdk/pkg/openfeature/testing"
+)
+
+// ctxFunction is a context based evaluation callback
+var ctxFunction = func(this mp.InMemoryFlag, evalCtx openfeature.FlattenedContext) (
+	interface{}, openfeature.ProviderResolutionDetail) {
+
+	defaultValue := this.Variants[this.DefaultVariant]
+	defaultResolution := openfeature.ProviderResolutionDetail{
+		Reason:  openfeature.DefaultReason,
+		Variant: this.DefaultVariant,
+	}
+
+	// check for expected properties
+	fn, ok := evalCtx["fn"].(string)
+	if !ok {
+		return defaultValue, defaultResolution
+	}
+
+	if fn != "Sulisław" {
+		return defaultValue, defaultResolution
+	}
+
+	ln, ok := evalCtx["ln"].(string)
+	if !ok {
+		return defaultValue, defaultResolution
+	}
+
+	if ln != "Świętopełk" {
+		return defaultValue, defaultResolution
+	}
+
+	age, ok := evalCtx["age"].(int64)
+	if !ok {
+		return defaultValue, defaultResolution
+	}
+
+	if age != 29 {
+		return defaultValue, defaultResolution
+	}
+
+	customer, ok := evalCtx["customer"].(bool)
+	if !ok {
+		return defaultValue, defaultResolution
+	}
+
+	if customer != false {
+		return defaultValue, defaultResolution
+	}
+
+	return this.Variants["internal"], openfeature.ProviderResolutionDetail{
+		Reason:  openfeature.TargetingMatchReason,
+		Variant: "internal",
+	}
```

Comment:
I'm not sure this all needs to be so verbose, particularly because there's no specific logging for each assertion failure. I think you could just do the type assertion and equality check in a single step for each, no?

Reply:
Yeah, will rework this :)

Reply:
Done, reduced the complexity - https://github.com/open-feature/go-sdk/pull/192/files#diff-25f457711915ad116e8b162b1c338a4eeaf032cbe097e20d34de53b5ec4812d3R9-R35 :)